geode-core/src/test/java/org/apache/geode/internal/cache/execute/PRTransaction2DUnitTest.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.cache.execute;
+
+import static org.junit.Assert.assertEquals;
+import static org.assertj.core.api.Assertions.*;
+
+import org.apache.geode.cache.PartitionAttributes;
+import org.apache.geode.cache.PartitionAttributesFactory;
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.RegionFactory;
+import org.apache.geode.cache.RegionShortcut;
+import org.apache.geode.cache.TransactionException;
+import org.apache.geode.internal.cache.TXManagerImpl;
+import org.apache.geode.test.dunit.Host;
+import org.apache.geode.test.dunit.VM;
+import org.apache.geode.test.dunit.cache.internal.JUnit4CacheTestCase;
+import org.apache.geode.test.junit.categories.DistributedTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@Category(DistributedTest.class)
+public class PRTransaction2DUnitTest extends JUnit4CacheTestCase {
+  Host host = Host.getHost(0);
+  VM dataStore1 = host.getVM(0);
+  VM dataStore2 = host.getVM(1);
+
+  @Test
+  public void testSizeOpOnLocalRegionInTransaction() {
+    String regionName = "region";
+    String region2Name = "region2";
+    int totalBuckets = 2;
+    boolean isSecondRegionLocal = true;
+    setupRegions(totalBuckets, regionName, isSecondRegionLocal, region2Name);
+
+    dataStore1.invoke(() -> verifySizeOpTransaction(2, regionName, totalBuckets, region2Name,
```

Comment:
I might even just get rid of verifySizeOpTransaction and put the assert code from there as the lambda here: makes nice and explicit what the exceptional result of this test should be. Then the test below just calls directly the method that does the transaction (which could be renamed verifySizeOpTransaction to maximize clarity).